### PR TITLE
fix: install compatible chrome version for puppeteer

### PR DESCRIPTION
**Pull request by vladfrangu**
Locally, this seems to work, however a test run should be done before merging if possible! 🙏

### CODE REVIEW

File: node-puppeteer-chrome/puppeteer_utils.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * Downloads the closest version of Chrome available
+ * @param {import('puppeteer').Browser} browser
+ * @param {string} versionToCheck
+ */
+async function downloadClosestChromeInstaller(browser, versionToCheck) {
+	// Get the first 3 parts of the version
+	const relevantPart = versionToCheck.split('.').slice(0, 3).join('.');
+	const page = await browser.newPage();
+	// Lazy load node-fetch here, since it won't be present outside the download process
+	const fetch = require('node-fetch');
```
Comment (mnmkng) [inline node-puppeteer-chrome/puppeteer_utils.js:60]: Why not require at top? The performance gains from not requiring are negligible.

Comment (vladfrangu) [reply node-puppeteer-chrome/puppeteer_utils.js:60]: This file is shared between the tester and the downloader - the former of which won't have node-fetch present, thus causing an error to be thrown

File: node-puppeteer-chrome/puppeteer_download.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const compatibleChromeVersion = matchedCompatibilityVersions
+        .map(v => v.chrome)
+        .sort((a, b) => b.localeCompare(a))[0];
+
+    const buffer = await downloadClosestChromeInstaller(browser, compatibleChromeVersion);
+    await writeFile('/tmp/chrome.deb', buffer);
+}
+
+(async () => {
+	console.log('Downloading the latest Chrome compatible with the version of Puppeteer that is installed');
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply node-puppeteer-chrome/puppeteer_download.js:31]: Good point, will add better logging 👍

File: node-puppeteer-chrome/puppeteer_utils.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    const [actualMajor, actualMinor] = actual.split('.').map(n => Number(n));
+    return actualMajor === compMajor && actualMinor >= compMinor;
+}
+
+/**
+ * @param {import('puppeteer').Browser} browser
+ * @returns
+ */
+async function fetchCompatibilityVersions(browser) {
+	const page = await browser.newPage();
```
Comment (B4nan) [inline node-puppeteer-chrome/puppeteer_utils.js:43]: mixed tabs/spaces

**Review by B4nan — CHANGES_REQUESTED** [review]
looking good, just two small points, plus please improve the comment as Ondra suggeested.

let's ship this today

**Pull request merged**